src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/fetch_and_validate.tsx

```diff
@@ -111,6 +110,7 @@ export function fetchAndValidate$({
 
           ignoreValidations,
           ...fetchContext,
+          filters: getFetchContextFilters(fetchContext, useGlobalFilters),
```

Comment:
This fixes a bug where:

1. The user disables Use Global Filters, and then the control fetches new data with filters disabled
2. The user re-enables Use Global Filters, but the control doesn't fetch new data until the user makes a change to the filters elsewhere on the dashboard

When we mutated `fetchContext.filters`, it wouldn't replace the missing filters when only the `useGlobalFilters` subject emitted a new value.

This was probably a preexisting bug.